### PR TITLE
add device secret support for nautobot

### PR DESCRIPTION
**Pull request by jeffkala**
Related to https://github.com/nautobot/nornir-nautobot/pull/42

@chipn

### CODE REVIEW

File: nautobot_plugin_nornir/plugins/inventory/nautobot_orm.py [hunk 1]
```diff
@@ -183,7 +183,8 @@ def create_host(self, device, cred, params: Dict):
         # require password for now
         host["password"] = password
 
-        # TODO save secret in connection option
+        # Update to use secret if specified.  Netmiko only for initial implementation.
+        host["data"]["connection_options"]["netmiko"]["extras"]["secret"] = secret
```
Comment (itdependsnetworks) [inline nautobot_plugin_nornir/plugins/inventory/nautobot_orm.py:187]: This was all that was required to get this working?

Comment (chipn) [reply nautobot_plugin_nornir/plugins/inventory/nautobot_orm.py:187]: Yes.

Comment (itdependsnetworks) [reply nautobot_plugin_nornir/plugins/inventory/nautobot_orm.py:187]: ok, can we not add napalm as well on this initial implementation?

Comment (chipn) [reply nautobot_plugin_nornir/plugins/inventory/nautobot_orm.py:187]: This line should work for Napalm:

`host["data"]["connection_options"]["napalm"]["extras"]["optional_args"]["secret"] = secret`

We'll need to test though.

Comment (jeffkala) [reply nautobot_plugin_nornir/plugins/inventory/nautobot_orm.py:187]: Thanks will get it added and see if I have a decent way to test.

**Comment by itdependsnetworks** [comment]
Can you get tests to pass, then i'll merge

**Comment by jeffkala** [comment]
> Can you get tests to pass, then i'll merge

@itdependsnetworks do you want me to fix the tests in a different PR?  This is a bug/issue with black in the new python 3.9 version that is being used by the github runners.  The fix is pinning python version and pinning black version.  So its outside the scope of this PR.

**Pull request merged**